feat: Add loading params from docusuarus.config.js

aloisklink opened the pull request:
Hi @KohheePeace,

I want to make a docusuarus template for my projects and put this package into them, but it is a bit difficult to change the `baseUrl`/`firstDocPagePath` in multiple places, so I made a new command to load them from `docusaurus.config.js`.

The command can be run like below, and I have add instructions into the README.md

```sh
npx docusaurus-pdf from-build-config
```

Let me know what you think :smile:

maxarndt commented on bin/index.js:
```diff
@@ -40,6 +40,28 @@ program
   });
 
+program
+  .command('from-build-config [siteDir]')
```
This look different than at `from-build`:
https://github.com/KohheePeace/docusaurus-pdf/blob/d90109638a5c98e6a956af3cf550c01b176853cf/bin/index.js#L26

It seems you know how to deal with commander.js. Please feel free to fix the command of `from-build` to have an uniform API across the different commands.

aloisklink replied:
Ah, yes, that is my fault :smile: I was trying to copy the `docusaurus build` API, but I think it makes more sense to copy the `from-build` API.

I have changed `from-build` slightly to use commander.js's default option value, and specified that `--output-file` must have an argument. I'm not sure if I should change the API anymore, since there are a few people using this package according to npm, and changing the API might break their scripts.

I've also changed `from-build-config` to match `from-build` more. Now `dirPath` is an argument, and `siteDir` is an option.

maxarndt replied:
You're right! I totally forgot that there are already some users using this project! We should avoid changing the API unless it is absolutely necessary. 😊

Thank's for the fix with the `--output-file` argument 👌

aloisklink commented:
Hi @maxarndt, I've adjusted the API of `from-build-config` to more closely match `from-build`. I don't want to change `from-build`, just in case there are people out there using the old version.

If you want me to, I'm also happy to write some tests with a testing framework.

maxarndt commented:
@aloisklink writing some tests for this project would be very good and essential. I have had this in my head for a long time and would also join in on the occasion.

A bug has also crept in that would not have existed with simple unit tests, as documented in #11.

So please feel free to add some tests 😊

aloisklink commented:
Hi @maxarndt,

I've added some basic tests with the jest framework that runs:

```sh
docusaurus-pdf https://v2.docusaurus.io/docs/2.0.0-alpha.56/docusaurus.config.js
```

and runs `from-build`/`from-build-config` on small docusaurus website in `tests/test-website`.

I've also tested and closed #11.

Finally, I've added a GitHub Action Continuous Integration to run the tests automatically. It should look something like this https://github.com/aloisklink/docusaurus-pdf/runs/811507203 once it's been merged.

I can also add an eslint/prettier config, but I think that should be in a separate pull-request, since it will most likely need to make a lot of changes.

Let me know what you think :smile:

KohheePeace commented:
@aloisklink Thanks for your work! 

> From my point of view this can be merged.

=> Same 👍  @maxarndt Could you squash and merge this PR?

> What do you think @KohheePeace? What about adding @aloisklink as collaborator? This would make his work a little bit more convenient 🙂

=> Sure sure @aloisklink  I just sent a invitation, please determine to accept it or not 👍 

@maxarndt I also want to add you as npm collaborators, what is your username of npm ??

aloisklink commented:
Awesome! Thank you @KohheePeace, I've accepted the invite :) Looking forwards to helping out

KohheePeace commented:
@aloisklink what is your npm https://www.npmjs.com/ username ? I will also add you as npm collaborators, and want you to publish this PR.

aloisklink commented:
Hi @KohheePeace, I'm happy to publish it. I can even write a Github Action script so we can publish it directly from GitHub. My npm username is [aloisklink](https://www.npmjs.com/~aloisklink).

KohheePeace commented:
>  I can even write a Github Action script so we can publish it directly from GitHub.

=> @aloisklink awesome! 😀  Just sent invitation 👍

aloisklink commented:
Great, v1.20 has been published :)

I haven't made a GitHub Action for automatic `npm publish`, since only the admin can do it. But I think maybe it is a bad idea, since currently with npm, when you upload a key, it will be a key for your entire account, not just one package, see https://help.github.com/en/actions/language-and-framework-guides/publishing-nodejs-packages#publishing-packages-to-the-npm-registry

maxarndt commented:
Welcome on board @aloisklink!

@KohheePeace my npm username is maxarndt 😊

KohheePeace commented:
@maxarndt invited you ! 😀 thanks!

@aloisklink 
> I haven't made a GitHub Action for automatic npm publish, since only the admin can do it. But I think maybe it is a bad idea, since currently with npm, when you upload a key, it will be a key for your entire account, not just one package, see

=> 👍
